myapp/app/controllers/tasks_controller.rb

```diff
@@ -6,10 +6,11 @@ def index
     @task_columns_with_sorting_direction = task_columns_with_sorting_direction
 
     @tasks = @current_user.tasks.status(params[:status]).name_contain(params[:name]).tag_contain(params[:tag_id])
-    @tasks = @tasks.order("#{params[:sort_key]} #{sort_direction}") if params[:sort_key].present?
 
     # 特定のタグに紐づくタスクに絞り込んだ後、それぞれのタスクに紐づくタグをすべて表示したいので、再度 Task のクエリを実行する
-    @tasks = Task.where(id: @tasks.map{|t| t.id}).includes(:tags).page(params[:page])
+    @tasks = Task.where(id: @tasks.map { |t| t.id }).includes(:tags).page(params[:page])
+
+    @tasks = @tasks.order("#{params[:sort_key]} #{sort_direction}") if params[:sort_key].present?
```

Comment:
[bugfix] After [this change](https://github.com/Fablic/training/pull/1530/files#diff-e75236d23a45a89b1d77b043070dcb62d0202e7897322cbef5381cc0e71d4339R8-R12), sorting function did not work.